src/Microsoft.Azure.WebJobs.Host/JobHostConfiguration.cs

```diff
@@ -164,6 +167,17 @@ public string StorageConnectionString
         /// </summary>
         public TimeSpan? FunctionTimeout { get; set; }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether function invocations will timeout when
+        /// a <see cref="FunctionTimeout"/> is specified and a debugger is attached. False by default.
+        /// </summary>
+        public bool TimeoutWhileDebugging { get; set; }
```

Comment:
How about moving all 3 of these related settings into a new FunctionTimeoutConfiguration class, and exposing it via JobHostConfiguration.FunctionTimeout property?

Then I'd have the new config class have properties: DefaultTimeout, ThrowOnTimeout and IgnoreWhileDebugging